### PR TITLE
Browser testing for examples and main page. Closes #656

### LINKED ISSUE #656: Add browserstack testing to the main page and examples
We should have tests checking for js error in these as well.

### PR DESCRIPTION
**Pull request by bjedrzejewski**
Refactored the test code and testing examples and main page as well.

### CODE REVIEW

File: site/src/examples/basecoin/index.js [hunk 1]
```diff
@@ -308,7 +308,6 @@
                 }
                 return sum;
             }, 0);
-            console.log('avg', sum / (frameTimings.length - 1));
```
Comment (chrisprice) [inline site/src/examples/basecoin/index.js]: Opps...

Comment (bjedrzejewski) [inline site/src/examples/basecoin/index.js]: Test picked it up :)

File: webdriver-tests/siteConsoleSpec.js [hunk 2]
```diff
@@ -1,36 +1,49 @@
...
             .getAttribute('.nav-stacked a', 'href')
             .then(function(attr) {
+                urls = attr;
+                return testNextPage(urls, done);
+            });
+    });
+
+    it('site examples contains no logs / errors', function(done) {
+        var urls = [];
+        return browser.url('/examples')
```
Comment (chrisprice) [inline webdriver-tests/siteConsoleSpec.js]: Could this be extracted out into a function e.g. getLinks(baseUrl, cssSelector)?

Comment (bjedrzejewski) [inline webdriver-tests/siteConsoleSpec.js]: Sure, although I suggest keeping it local to the file as it is very specific to our current layout / website

Comment (chrisprice) [inline webdriver-tests/siteConsoleSpec.js]: Agreed.

**Comment by bjedrzejewski** [comment]
@chrisprice refactored the code a bit more

**Comment by bjedrzejewski** [comment]
Wait with merge I need to fix one more thing

**Comment by bjedrzejewski** [comment]
@chrisprice This is now working and ready to merge / review futher

**Comment by chrisprice** [comment]
:+1: Let's see if it works!

**Pull request merged**